packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -100,14 +141,27 @@ export default class SaytDriverPlugin implements Plugin {
    * @returns A promise from the Sayt API that has been reformatted
    * with the passed callback.
    */
-  sendSaytApiRequest(query: string, config: QueryTimeAutocompleteConfig): Promise<string[]> {
+  sendAutocompleteApiRequest(query: string, config: QueryTimeAutocompleteConfig): Promise<string[]> {
     return this.core.sayt.autocomplete(query, config).then(this.autocompleteCallback);
   }
 
+  /**
+   * Sends a request to the Search API with the given query and config object.
+   *
+   * @param query The search term to send.
+   * @param config Extra query-time configuration to customize the Search request.
+   * @returns A promise from the Search API that has been reformatted
+   * with the passed callback.
+   */
+  sendSearchApiRequest(query: string, config: QueryTimeAutocompleteConfig): Promise<ProductsResponseSection> {
+    const allConfig = { ...this.defaultSearchConfig, ...config };
+    return this.core.search.search({ query, ...allConfig }).then(this.searchCallback);
```

Comment:
I don't think `allConfig` enhances readability that much, so I think it can be inlined. As well, the default should be the first in the whole object so that any `query` parameters that might be given as default gets overridden.

```suggestion
    return this.core.search.search({ ...this.defaultSearchConfig, query, ...config }).then(this.searchCallback);
```